## Objective

The objective of this notebook is to import the dataset, examine its structure, evaluate data quality, and identify any issues that need to be addressed before analysis or modeling.

In [1]:
import pandas as pd
import numpy as np

In [2]:
df = pd.read_csv("../data/music_data_raw.csv")

In [3]:
df.columns

Index(['unit_short', 'subject_ID', 'sex', 'age', 'years_of_education', 'group',
       'country', 'unit', 'subject_ID of the matched participant', 'wais3',
       'wais4', 'waisvocabulary', 'raven', 'digit_span', 'spatial_span',
       'melody_span', 'nback', 'PROMSMELODYSCORE', 'PROMSTUNINGSCORE',
       'PROMSBEATSCORE', 'PROMSSPEEDSCORE', 'PROMSTOTALSCORE',
       'PROMSTOTALSCORENOMELODY', 'eBMRQ_MusicSeeking',
       'eBMRQ_EmotionEvocation', 'eBMRQ_MoodRegulation', 'eBMRQ_SensoryMotor',
       'eBMRQ_SocialReward', 'eBMRQ_MusicalAbsorption', 'eBMRQ_TOT',
       'BMRQ_TOT', 'gold_msi_active_engagement',
       'gold_msi_perceptual_abilities', 'gold_msi_musical_training',
       'gold_msi_emotions', 'gold_msi_singing_abilities',
       'gold_msi_general_sophistication', 'bfi_extraversion',
       'bfi_agreeableness', 'bfi_conscientiousness',
       'bfi_negative_emotionality', 'bfi_open_mindedness',
       'ses_nuclear_family', 'AddMusicTraining[years]',
       'lastMusicActivity[y

In [4]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 1438 entries, 0 to 1437
Data columns (total 49 columns):
 #   Column                                 Non-Null Count  Dtype  
---  ------                                 --------------  -----  
 0   unit_short                             1438 non-null   str    
 1   subject_ID                             1438 non-null   str    
 2   sex                                    1438 non-null   str    
 3   age                                    1438 non-null   int64  
 4   years_of_education                     1438 non-null   int64  
 5   group                                  1438 non-null   str    
 6   country                                1438 non-null   str    
 7   unit                                   1438 non-null   str    
 8   subject_ID of the matched participant  1274 non-null   str    
 9   wais3                                  90 non-null     float64
 10  wais4                                  1336 non-null   float64
 11  waisvocabulary 

- Dataset contains 1,438 observations.

- Dataset contains 49 variables.

- Both numerical and categorical variables are present.

- Memory measures include digit_span, spatial_span and melody_span.

- Further investigation of missing values is required.